tests/kernel/mem_pool/sys_mem_pool/src/main.c

```diff
@@ -19,10 +20,35 @@
 #define BLK_SIZE_EXCLUDE_DESC (BLK_SIZE_MIN - 16)
 #define BLK_ALIGN BLK_SIZE_MIN
 
+K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+__kernel struct k_thread tdata;
 
 K_MUTEX_DEFINE(pool_mutex);
 SYS_MEM_POOL_DEFINE(pool, &pool_mutex, BLK_SIZE_MIN, BLK_SIZE_MAX,
 		    BLK_NUM_MAX, BLK_ALIGN, .data);
+K_SEM_DEFINE(end_sema, 0, 1);
+
+#ifndef CONFIG_USERSPACE
+void test_sys_mem_pool_assign(void);
+#endif
+
+/**
+ * @see k_malloc(), k_free()
+ */
+#ifndef CONFIG_USERSPACE
+static void thread_handler(void *p1, void *p2, void *p3)
+{
+	char *ptr;
+
+	k_thread_system_pool_assign(k_current_get());
+	ptr = k_malloc(128);
```

Comment:
k_malloc and k_thread_system_pool_assign have nothing to do with each other, what exactly are we testing here?

k_thread_system_pool_assign assigns a memory pool for z_thread_malloc, not k_malloc

also why is this test have a ifndef CONFIG_USERSPACE around it??

Reply:
In this test case we are trying to validate k_thread_system_pool_assign() api.
We are assigning system heap resource to the current thread. And later trying to allocate and free memory from the system heap memory. 

As you suggested I have removed k_malloc and used z_thread_malloc() instead.

I was facing some shippable issue with qemu_x86, to resolve that issue I added "ifndef CONFIG_USERSPACE". I removed "ifndef CONFIG_USERSPACE" and not getting any shippable issue now.